cmd/ipfs/main.go

```diff
@@ -476,11 +479,13 @@ func writeHeapProfileToFile() error {
 }
 
 // listen for and handle SIGTERM
-func (i *cmdInvocation) setupInterruptHandler() {
+func (i *cmdInvocation) setupInterruptHandler() *sync.WaitGroup {
```

Comment:
returning the wg is not idiomatic-- they're not used like that generally. if this really needs to be the flow, can return an `io.Closer` that effectively calls `wg.Wait()`.